src/utils/helpers.js

```diff
@@ -58,6 +58,22 @@ export function parseJSON(response) {
     })
   }
 
+  if (response.status === 429) {
+    // Only the body of the response is passed back so to catch a 429
+    // we fill the JSON body with some error details to give information
+    // to the consumer
+    const error = {
+      errors: [{ status: 429 }]
+    }
+    return new Promise(resolve => {
```

Comment:
@craigtweedy We can tidy this whole function up really - no need for 3 separate returns based on `status` logic.

Reply:
I agree - was mostly to fit into the style already there but I will update this entire function.

Reply:
How about this? 

```js
export function parseJSON(response) {

  return new Promise(resolve => {
    response.text().then(body => {

      let json = '{}'

      if (body !== '') {
        json = JSON.parse(body)
      } 
      
      if (!response.ok) {
        json = { errors: [{ status: response.status }] }
      }

      resolve({
        status: response.status,
        ok: response.ok,
        json
      })
    })
  })
}
```

Reply:
@craigtweedy Looks better, but not sure why `text()` is needed in this instance? Maybe I'm misunderstanding.

```js
export function parseJSON(response) {
  return new Promise(resolve => {
    const { status, ok } = response

    response.json().then(payload => 
      let json = payload

      if (!ok) {
        json = { errors: [{ status }] }
      }

      if (status === 204) {
        json = '{}'
      }

      resolve({
        status,
        ok,
        json
      })
    )
  }) 
}
```

Reply:
`response.json()` throws an error if the body is empty - which it is in a couple of situations like when we hit the rate limit. Your example above would still cause the SDK to bail out when the rate limit is hit. 

For this reason, I figured parsing as text and then filling in the empty body would be the path of least resistance.